builder/bundles.go

```diff
@@ -493,15 +547,79 @@ func installBundleToFull(packagerCmd []string, buildVersionDir string, bundle *b
 			rpmList = append(rpmList, p)
 		}
 
+		startTime := time.Now()
 		// Retry RPM downloads to avoid timeout failures due to slow network
 		_, err = downloadRpms(packagerCmd, rpmList, baseDir, downloadRetries)
 		if err != nil {
 			return err
 		}
+		fmt.Println("download time : ", time.Since(startTime).String())
+		elapsedTime = elapsedTime + time.Since(startTime)
+
+		path := baseDir + "/var/cache/yum/clear/"
+		newPath, err := filepath.Glob(filepath.Join(path, "clear-*"))
+		if err != nil {
+			fmt.Print("cant find path to rpm")
+			return err
+		}
+		if len(newPath) > 0 {
+			newPath[0] = newPath[0] + "/packages"
+		} else {
+			return fmt.Errorf("cant find rpms packages")
+		}
+
+		var wg sync.WaitGroup
+
+		if len(rpmList) < numWorkers {
+			numWorkers = len(rpmList)
+		}
+
+		wg.Add(numWorkers)
+		rpmCh := make(chan string)
+		errorCh := make(chan error, numWorkers)
+		rpmWorker := func() {
+			for rpm := range rpmCh {
+				e := extractRpm(baseDir, rpm)
+				if e != nil {
+					errorCh <- e
+					break
+				}
+			}
+			wg.Done()
+		}
+
+		for i := 0; i < numWorkers; i++ {
+			go rpmWorker()
+		}
 
-		args := merge(packagerCmd, "--installroot="+baseDir, "install")
-		args = append(args, rpmList...)
-		_, err = helpers.RunCommandOutputEnv(args[0], args[1:], []string{"LC_ALL=en_US.UTF-8"})
+		// feed the channel
+		for _, rpm := range rpmList {
+			rpmFull := rpm + "*.rpm"
+			fullPath, err := filepath.Glob(filepath.Join(newPath[0], rpmFull))
+			if err != nil {
+				fmt.Println("cant find rpm path")
+				return err
+			}
+			for _, i := range fullPath {
+				if rpmMap[i] == 1 {
+					continue
+				}
+				rpmMap[i] = 1
+				select {
+				case rpmCh <- i:
+				case <-errorCh:
+					// break as soon as there is a failure.
+					break
+				}
+			}
+		}
+		close(rpmCh)
+		wg.Wait()
+		// an error could happen after all the workers are spawned so check again for an
+		// error after wg.Wait() completes.
+		if len(errorCh) > 0 {
+			err = <-errorCh
```

Comment:
Nothing sets err above this line, so you can call return here directly and delete the `if err != nil` below.